api/src/main/java/com/datastrato/gravitino/rel/SupportsSchemas.java

```diff
@@ -94,20 +94,22 @@ Schema createSchema(NameIdentifier ident, String comment, Map<String, String> pr
    * @param changes The metadata changes to apply.
    * @return The altered schema.
    * @throws NoSuchSchemaException If the schema does not exist.
+   * @throws UnsupportedOperationException If the catalog does not support altering a schema.
    */
-  Schema alterSchema(NameIdentifier ident, SchemaChange... changes) throws NoSuchSchemaException;
+  Schema alterSchema(NameIdentifier ident, SchemaChange... changes)
+      throws NoSuchSchemaException, UnsupportedOperationException;
```

Comment:
We typically use `IllegalArgumentException` for unsupported changes. please see:

```java
  /**
   * Apply the {@link TableChange change} to a table in the catalog.
   *
   * <p>Implementations may reject the change. If any change is rejected, no changes should be
   * applied to the table.
   *
   * @param ident A table identifier.
   * @param changes Table changes to apply to the table.
   * @return The updated table metadata.
   * @throws NoSuchTableException If the table does not exist.
   * @throws IllegalArgumentException If the change is rejected by the implementation.
   */
  Table alterTable(NameIdentifier ident, TableChange... changes)
      throws NoSuchTableException, IllegalArgumentException;
```

I think we should unify them.

Reply:
Is `UnsupportedOperationException` more clear and detailed than `IllegalArgumentException`?

Reply:
OK,  I will undo this change here.